apps/api/src/app/inbox/usecases/update-preferences/update-preferences.usecase.ts

```diff
@@ -55,46 +55,16 @@ export class UpdatePreferences {
       }
     }
 
-    const userPreference: SubscriberPreferenceEntity | null = await this.subscriberPreferenceRepository.findOne(
-      this.commonQuery(command, subscriber)
-    );
-    if (!userPreference) {
-      await this.createUserPreference(command, subscriber);
-    } else {
-      await this.updateUserPreference(command, subscriber);
-    }
+    await this.updateSubscriberPreference(command, subscriber);
```

Comment:
As said above, we treat this as an upsert now. `GetSubscriberPreferences` always returns a valid preference set now (falling back to default when nothing has been persisted), so there's no need to perform an extra lookup now before persisting the data.